### DISCUSSION TITLE
Georeferencing graphic files to create maps

### DISCUSSION THREAD

**Discussion by john-a-p** [osmandapp/OsmAnd#18579] [Ideas]
I use OsmAnd a lot, but here's a feature I'd like to have that the program doesn't provide. That would be the ability to create maps from graphic files. I know it can be done with techniques on the "Raster maps" page, but it's complicated. I doubt if many users can deal with sqlitedb, and to be honest, I don't think something so simple should be so difficult. I probably wouldn't post about it, but I just found an Android app that does exactly what I want, and it's extremely simple to use. I don't think it would be very challenging to add something similar to OsmaAnd.

The app is called Custom Maps, and it's free on the Play Store. What it does is let you select a graphic file, a JPG or PDF. It opens that up, and you designate points on the screen, then it starts Google Maps and you locate the same point there. You can enter from 2 to 4 points for each map. From there, it creates a new georeferenced version of the file, and then you can load it into the program as a map and see your location on it. That's all it does, with very few other features. But it's enough, and obviously a user could run OsmAnd at the same time, and have access to all the other features there. It seems to me that Custom Maps is using a very simple process to create and show a georeferenced image as a map. It's easy to see that if OsmAnd could do this, there wouldn't be any need to use Google Maps; the user would just find the desired location (to match a location on the selected graphic image) on whatever map Osmand is set to display. Alternatively (Custom Maps doesn't allow this, but it would be useful) the user could simply enter a latitude and longitude. This would let a user add a custom map without any specialized knowledge.

**Comment by mikehgentry** [comment]
I've often thought it'd be neat to be able to walk up to a sign with a map on it (e.g. in a park), take a photo of it from within OsmAnd, and turn it into an underlay / overlay.

Might also be useful for paper tourist maps, but they're often quite schematic (not to mention folded) so I imagine you'd need to identify a lot of common points to stretch them correctly.

**Comment by tommycrock** [comment]
I use Custom Maps and would appreciate the feature being integrated with OsmAnd. I've used it with the maps posted at park/zoo entrances and Wainwright maps. Oddly I find it often works best with only two points, four seems to end up distorting it more. Makes it easy to use though

**Reply by ERYpTION** [reply]
I've georeferenced several pdf-maps using QGIS and am surprised to hear that you get good results with only two points. As far as I remember in QGIS the minimum is five. Sometimes several more are needed if the image is not accurate. I must try the app.

**Reply by tommycrock** [reply]
Well, I may be misremembering about two rather than three, and I'm not saying great results (I wouldn't want to trace off them) but very serviceable for 'roughly where am I'. My memory is things got worse as I added more like five points - maybe there is some assumptions about the transformation which don't hold. 
It wraps the image and a kml file defining the tie points in a kmz so it can be shared, which is nice and simple.

**Reply by john-a-p** [reply]
One of the examples that Custom Maps shows was indeed a map of hiking trails at a conservation area somewhere. Another was a 17th-century map of Venice, which I thought was a pretty impressive demonstration of how versatile it is! But then again, it's doing exactly the same thing in both cases, so maybe it's not so very remarkable.

Two points seem like the bare minimum for a correlation of this kind. What it would do would be to stretch the graphic so that the distance between the points matches, and skew it so that the angle relative to a north-south meridian also matches. That'll work well if you accurately locate the points on accurate maps, but obviously if the graphic is just a sketch, you'll get "sketchy"accuracy from it.

**Reply by ERYpTION** [reply]
I have now tried it myself with one pdf-map and it worked well. I only used two points but for every point you rotate the map to align it with the reference (Gmap). Very simple and the result looked sufficiently accurate. Would for certain be a great feature / plugin for OsmAnd although perhaps not something in great demand from the average casual user.

**Comment by vshcherb** [comment]
It's possible to do on PC via QGis for example https://www.youtube.com/watch?v=jKLBFddpTGI and create raster tiles, though I agree do in the program would be nicer
